2 dz/6/main.cpp

```diff
@@ -0,0 +1,144 @@
+#include <iostream>
+#include <cstdlib>
+#include <ctime>
+#include <cmath>
+
+using namespace std;
+
+void num_gen(int *hidden_number, int size)
+{
+    srand(time(NULL));
+    for (int i = 0; i < size; i++)
+    {
+        hidden_number[i] = rand() % 10;
+        if (hidden_number[0] == 0)
+        {
+            i = -1;
+            break;
+        }
+        if (i > 0)
+        {
+            for (int j = i - 1; j >= 0; j--)
+            {
+                if (hidden_number[i] == hidden_number[j])
+                {
+                    i = -1;
+                    break;
+                }
+            }
+        }
+    }
+}
+
+int nob(int *hidden_number, int *new_number, int size)
+{
+    int n = 0;
+    for (int i = 0; i < size; i++)
+    {
+        if (hidden_number[i] == new_number[i])
+        {
+            n++;
+        }
+    }
+    return n;
+}
+
+int noc(int *hidden_number, int *new_number, int size)
+{
+    int n = 0;
+    for (int i = 0; i < size; i++)
+    {
+        for (int j = 0; j < size; j++)
+        {
+            if (i == j)
+            {
+                j++;
+            }
```

Comment:
`for (int j = 0; j < size && j != i; j++)` ?

Reply:
you can get j == size - 1, then do `j++`, then try `new_number[j]` in the next line. Oups... Invalid memory access. Invalid index.